skimage/graph/_graph_cut.py

```diff
@@ -277,34 +277,35 @@ def _ncut_relabel(rag, thresh, num_cuts, random_generator):
             #
             # It may make sense to a warning here; on the other hand segmentations
             # are not a ground truth, so this level of "noise" should be acceptable.
-            return
-
-        d2 = d.copy()
-        # Since d is diagonal, we can directly operate on its data
-        # the inverse of the square root
-        d2.data = np.reciprocal(np.sqrt(d2.data, out=d2.data), out=d2.data)
-
-        # Refer Shi & Malik 2001, Equation 7, Page 891
-        A = d2 @ (d - w) @ d2
-        # Initialize the vector to ensure reproducibility.
-        v0 = random_generator.random(A.shape[0])
-        vals, vectors = linalg.eigsh(A, which='SM', v0=v0, k=min(100, m - 2))
-
-        # Pick second smallest eigenvector.
-        # Refer Shi & Malik 2001, Section 3.2.3, Page 893
-        vals, vectors = np.real(vals), np.real(vectors)
-        index2 = _ncut_cy.argmin2(vals)
-        ev = vectors[:, index2]
-
-        cut_mask, mcut = get_min_ncut(ev, d, w, num_cuts)
-        if mcut < thresh:
-            # Sub divide and perform N-cut again
-            # Refer Shi & Malik 2001, Section 3.2.5, Page 893
-            sub1, sub2 = partition_by_cut(cut_mask, rag)
-
-            _ncut_relabel(sub1, thresh, num_cuts, random_generator)
-            _ncut_relabel(sub2, thresh, num_cuts, random_generator)
-            return
+            pass
+
+        else:
```

Comment:
I don't know this code - bit I think the change has the effect of running `_label_all(rag, 'ncut label')` when in denerate case.  I wonder whether there is a way of refactoing to make the logic cleare?.   For example:

```python
    # Comments about nnz == 0 case
    if m > 2 and (d != w).nnz > 0:
        # Logic for non-degenerate case.
```

or similar?

Reply:
Done